internal/httpproxy/http_proxy.go

```diff
@@ -146,8 +149,12 @@ func (p *ProxyConn) authenticate() error {
 		return err
 	}
 
+	if hasMetrics {
+		originalMetricsConn.setConnectionCategory(req)
```

Comment:
This method seems unnecessary coupling (why does a TCP connection need to know about HTTP request?). Would setting the metric label explicitly simpler here e.g.
```go
if isHealthCheckRequest(req) {
  originalMetricsConn.connectionCategory = connectionCategoryHealth
  // other code
}
originalMetricsConn.connectionCategory = connectionCategoryProxy
```
?